aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -17,34 +17,54 @@
+// 2. Use the admin_account to set up the collection's treasury address (where the minting fee will go)
+// set_treasury(&admin_account, signer::address_of(treasury_account));
+// 3. Set up the source and destination collection configs using the admin account. Note that here we are setting up both
+// the source and destination collection configs at the same time.
+// set_collection_config_and_create_collection()
+// 4. Set up the public minting and reveal config
+// set_public_minting_and_reveal_config()
+// 5. (Optional, do this if your project has a whitelist)
+// Set up the whitelist config in order. Because we're setting up a 2-stage whitelist project,
+// we want to call set_up_whitelist_stage() and add_to_whitelist() two times - first time
+// to set up the first stage, and the second time to set up the second stage.
+// set_up_whitelist_stage(&admin_account, start_time, end_time, price, 0); // setting up the first stage with 0 being the stage index (stage is 0-indexed)
+// add_to_whitelist(&admin_account, whitelist_address, mint_limit, 0); // adding whitelisted addresses to the first stage
+// set_up_whitelist_stage(&admin_account, start_time, end_time, price, 1); // setting up the second stage with 1 being the stage index
+// add_to_whitelist(&admin_account, whitelist_address, mint_limit, 1); // adding whitelisted addresses to the second stage
+// Currently, there's no way to reset the whitelist config.
+// 6. Add token uris for the destination collection
+// add_tokens()
+//
+// ##### Minting #####
+// 1. Mint a source certificate token from the source collection
+// mint_source_certificate()
+// 2. Exchange the source certificate token to a destination token
+// exchange()
 module post_mint_reveal_nft::minting {
-    use std::error;
-    use std::signer;
-    use std::string::{Self, String, utf8};
-    use std::vector;
-
     use aptos_framework::account::{Self, SignerCapability, create_signer_with_capability};
     use aptos_framework::aptos_coin::AptosCoin;
     use aptos_framework::coin;
     use aptos_framework::event::{Self, EventHandle};
-    use aptos_framework::resource_account;
     use aptos_framework::timestamp;
-    use aptos_token::token::{Self, TokenMutabilityConfig, create_token_mutability_config, create_collection, create_tokendata, TokenId, TokenDataId};
-    use post_mint_reveal_nft::bucket_table::{Self, BucketTable};
+    use aptos_token::token::{Self, TokenMutabilityConfig, create_token_mutability_config, create_collection, create_tokendata, TokenId};
     use post_mint_reveal_nft::big_vector::{Self, BigVector};
-    use std::bcs;
+    use post_mint_reveal_nft::bucket_table::{Self, BucketTable};
+    use post_mint_reveal_nft::whitelist;
 
+    use std::bcs;
+    use std::error;
+    use std::signer;
+    use std::string::{Self, String, utf8};
+    use std::vector;
 
     /// NFTMintConfig stores relevant information and events of this module.
     struct NFTMintConfig has key {
-        admin: address,
```

Comment:
FYI since we're heavily modifying these example contracts like this, anyone who previously deployed this will no longer be able to use this to do compatible upgrades. We might need to think about a process to version these examples